middle_end/closure/closure.ml

```diff
@@ -63,7 +63,16 @@ let rec build_closure_env env_param pos = function
    contain the right names if the -for-pack option is active. *)
 
 let getglobal dbg id =
-  Uprim(P.Pread_symbol (Compilenv.symbol_for_global id), [], dbg)
+  assert (Ident.is_global_or_predef id);
```

Comment:
This code really doesn't belong here - almost exactly the same code needs to be added to `Opttoploop` for the same purpose: translating a global `Ident.t` to a linkage name. `Compilenv.symbol_for_global` used to do this, and `Compilenv` still needs to be consulted anyway, so we should just keep `Compilenv.symbol_for_global` after all (with this new implementation).